contracts/Assertion.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { AssertionStorage } from "./storage/AssertionStorage.sol";
+import { Hub } from "./Hub.sol";
+import { AssertionStructs } from "./structs/AssertionStructs.sol";
+
+contract Assertion {
+
+	Hub public hub;
+	AssertionStorage public assertionStorage;
+
+	constructor(address hubAddress) {
+		require(hubAddress != address(0));
+
+		hub = Hub(hubAddress);
+		assertionStorage = AssertionStorage(hub.getContractAddress("AssertionStorage"));
+	}
+
+	modifier onlyAssetContracts() {
+        _checkAssetContract();
+        _;
+    }
+
+	function createAssertion(
+		bytes32 assertionId,
+		address issuer,
+		uint128 size,
+		uint32 triplesNumber,
+		uint96 chunksNumber
+	)
+		external
+		onlyAssetContracts
+	{
+		AssertionStorage ans = assertionStorage;
+
+		require(assertionId != bytes32(0), "Assertion ID cannot be empty");
+		require(!ans.assertionExists(assertionId), "Assertion already exists");
```

Comment:
We can also check issuer here to avoid duplicates
```suggestion
		require(!ans.assertionExists(assertionId) && (issuer == ans.getAssertionIssuer(assertionId)), "Assertion already exists");
```

Reply:
The problem is however that two different issuers cannot create a same assertion
this is in the AssertionStorage contract

`// assertionId => Assertion
mapping(bytes32 => AssertionStructs.Assertion) assertions;`

Lets have this as a TODO so we can solve it in the next PR @u-hubar